include/optimization/lp_problem.h

```diff
@@ -0,0 +1,237 @@
+//
```

Comment:
maybe `lp_solver` is a better name?

Reply:
It does sound better! Because this will affect future merges with different branches and probably git won't be able to keep track of the renamed files / classes, should we do it later?

Reply:
I do not see any reason `git` will not keep track of this change in other branches if you `merge` them with this one